libpimeval/src/pimSim.cpp

```diff
@@ -751,23 +751,138 @@ pimSim::pimPopCount(PimObjId src, PimObjId dest)
   return m_device->executeCmd(std::move(cmd));
 }
 
-template <typename T> bool
-pimSim::pimRedSum(PimObjId src, T* sum)
-{
-  pimPerfMon perfMon("pimRedSum");
+//! @brief  Min reduction operation
+bool pimSim::pimRedMin(PimObjId src, void* min, uint64_t idxBegin, uint64_t idxEnd) {
+  if (idxBegin != idxEnd && idxBegin < idxEnd) pimPerfMon perfMon("pimRedMinRanged");
+  else pimPerfMon perfMon("pimRedMin");
```

Comment:
```suggestion
  std::string tag = (idxBegin != idxEnd && idxBegin < idxEnd) ? "pimRedMinRanged" : "pimRedMin";
  pimPerfMon perfMon(tag);
```
Please note that pimPerfMon is an on-stack object and its ctor-to-dtor lifetime needs to cover entire operation.